nvtabular/io.py

```diff
@@ -918,21 +921,31 @@ def to_ddf(self, columns=None):
             return self._ddf[[columns]]
         return self._ddf
 
+    def make_iter(self, columns=None, indices=None):
+        if isinstance(columns, str):
+            columns = [columns]
+
+        return DataFrameIter(self._ddf, columns=columns, indices=indices)
+
     @property
     def num_rows(self):
         return len(self._ddf)
 
 
 class DataFrameIter:
-    def __init__(self, ddf, columns=None):
+    def __init__(self, ddf, columns=None, indices=None):
+        self.indices = (
+            indices if isinstance(indices, list) and len(indices) > 0 else range(ddf.npartitions)
```

Comment:
cant do this... bool not available for list type, it causes error saying you need to check any or all. So it make work with the range but if we give an actual list of values it will fail. Much cleaner this way... Originally wanted it like that.

Reply:
specifically something like this bubble:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

Reply:
good point, I hadn't considered c- maybe should just use if not None instead ?

```suggestion
        self.indices = indices if indices is not None else range(ddf.partitions)
```

Reply:
This method causes the following error:
`'IndexCallable' object cannot be interpreted as an integer`